httpx/models.py

```diff
@@ -710,7 +710,8 @@ def content(self) -> bytes:
             if hasattr(self, "_raw_content"):
                 raw_content = self._raw_content  # type: ignore
                 content = self.decoder.decode(raw_content)
-                content += self.decoder.flush()
```

Comment:
This line ought to stay the same. If there's an issue with a decoder erroring at the point of `flush()` being called against an empty buffer, then let's fix it on the decoder implementation.

Reply:
I couldn't figure out how to handle it in decoder. Is this make sense?

```
def flush(self) -> bytes:
        try:
            if hasattr(self.decompressor, "finish"):
                self.decompressor.finish()
            return b""
        except brotli.error as exc:  # pragma: nocover
            if str(exc) == "Decompression error: incomplete compressed stream.":
                return b""
            raise DecodingError from exc
```

Reply:
it comes from [this line](https://github.com/python-hyper/brotlipy/blob/ffddf2ea5adc584c8c353d246bb1077b7e781b63/src/brotli/brotli.py#L463). we can't use `BrotliDecoderIsFinished`. it is not public

Reply:
I think we can implement this by just having a flag that tracks whether the decoder actually received any data and if not just return `b""`.

Reply:
`def decode(self, data: bytes, with_flush: bool=False) -> bytes`